scripts/e2e-test-op-commands.py

```diff
@@ -0,0 +1,137 @@
+    "0x4641df4a962071e12719d8c8c8e5ac7fc4d97b927346a3d7a335b1f7517e133c"
+)
+
+
+@click.group("e2e-op-commands")
+@click.argument(
+    "keystore-file",
+    type=click.Path(exists=True, dir_okay=False, readable=True),
+    required=True,
+)
+@click.argument("password", type=str, required=True)
+@click.argument("l1-rpc", type=str, required=True)
+@click.pass_context
+def cli(ctx: Any, keystore_file: Path, password: str, l1_rpc: str) -> None:
+    account = account_from_keyfile(keystore_file, password)
+    web3_l1 = make_web3(URL(l1_rpc), account)
+    ctx.ensure_object(dict)
+    ctx.obj["account"] = account
+    ctx.obj["web3_l1"] = web3_l1
+
+
+@cli.command("verify-portal-call")
+@click.argument(
+    "portal-address",
+    type=str,
+)
+@click.argument(
+    "messenger-address",
+    type=str,
+)
+@pass_args
+def verify_portal_call(
+    account: LocalAccount,  # pylint:disable=unused-argument
+    web3_l1: Web3,
+    portal_address: str,
+    messenger_address: str,
+) -> None:
+    portal_logs = web3_l1.eth.get_logs(
+        {"address": to_checksum_address(portal_address), "fromBlock": 0}
+    )
+    messenger_logs = web3_l1.eth.get_logs(
+        {"address": to_checksum_address(messenger_address), "fromBlock": 0}
+    )
+    assert any(WITHDRAWAL_FINALIZED_TOPIC in log["topics"] for log in portal_logs)
+    assert any(FAILED_RELAYED_MESSAGE_TOPIC in log["topics"] for log in messenger_logs)
+    print("Processed Block Number:", web3_l1.eth.block_number)
+
+
+@cli.command("set-chain-on-resolver")
+@click.argument(
+    "deployment-dir",
+    type=Path,
+    required=True,
+)
+@click.argument("l2-rpc", type=str)
+@pass_args
+def set_messenger_on_resolver(
+    account: LocalAccount, web3_l1: Web3, deployment_dir: Path, l2_rpc: URL
+) -> None:
+    web3 = make_web3(l2_rpc, account)
+    env_chain_id = os.getenv("SOURCE_CHAIN_ID")
+    assert env_chain_id is not None
+    target_chain_id = ChainId(int(env_chain_id))
+
+    with deployment_dir.joinpath("deployment.json").open("rt") as f:
+        deployment = json.load(f)
+
+    deployment_info = prepare_deployment_infos(deployment_dir, deployment["base_chain"])
+    l1_contracts = make_contracts(web3_l1, deployment_info)
+    l2_contracts = contracts_for_web3(web3, deployment_dir)
+
+    resolver = l1_contracts["Resolver"]
+    request_manager = l2_contracts["RequestManager"]
+    l1_messenger = l1_contracts["OptimismL1Messenger"]
+
+    transact(
+        resolver.functions.addRequestManager(
+            target_chain_id, request_manager.address, l1_messenger.address
+        )
+    )
+
+
+@cli.command("verify-messenger-call")
+@click.argument(
+    "portal-address",
+    type=str,
+)
+@click.argument(
+    "messenger-address",
+    type=str,
+)
+@pass_args
+def verify_messenger_call(
+    account: LocalAccount,  # pylint:disable=unused-argument
+    web3_l1: Web3,
+    portal_address: str,
+    messenger_address: str,
+) -> None:
+    block_number = int(os.environ["PROCESSED_BLOCK_NUMBER"])
```

Comment:
I'm not sure why this is called "processed" block number and not just "block number".

Reply:
changed